src/credentials/examples/GroupDataProviderExample.cpp

```diff
@@ -0,0 +1,654 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <credentials/GroupDataProvider.h>
+#include <lib/support/CHIPMem.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/Pool.h>
+#include <stdlib.h>
+#include <string.h>
+
+namespace chip {
+namespace Credentials {
+namespace {
+
+static constexpr size_t kNumFabrics         = CHIP_CONFIG_MAX_DEVICE_ADMINS;
+static constexpr size_t kEndpointEntriesMax = CHIP_CONFIG_MAX_GROUP_ENDPOINTS_PER_FABRIC;
+static constexpr size_t kStateEntriesMax    = CHIP_CONFIG_MAX_GROUPS_PER_FABRIC;
+static constexpr size_t kKeyEntriesMax      = CHIP_CONFIG_MAX_GROUPS_PER_FABRIC;
+static constexpr size_t kIteratorsMax       = CHIP_CONFIG_MAX_GROUP_CONCURRENT_ITERATORS;
+
+class StaticGroupsProvider : public GroupDataProvider
+{
+protected:
+    struct EndpointEntry : public GroupMapping
+    {
+        bool in_use = false;
+    };
+
+    struct KeysEntry : public KeySet
+    {
+        bool in_use = false;
+    };
+
+    struct Fabric
+    {
+        // Whether entry is allocated (true), or free to allocate (false)
+        bool in_use = false;
+        // Node-wide fabric index associated with the entry
+        chip::FabricIndex fabric_index;
+        // Index within the outer list, used to do index remapping
+        uint8_t internal_index;
```

Comment:
I think the fabric slot index will have the same bounds as fabric_index.  Wouldn't `chip::FabricIndex` then make sense here?

```suggestion
        chip::FabricIndex internal_index;
```

Reply:
Can an explanation be provided for the rejection of this feedback?

It's fine to not agree with me.  I would just like to understand the rationale.

Reply:
Answered elsewhere.  This was apparently stale code.

All good!